darwin/utils/utils.py

```diff
@@ -1005,6 +1005,10 @@ def split_video_annotation(annotation: dt.AnnotationFile) -> List[dt.AnnotationF
             for a in annotation.annotations
             if isinstance(a, dt.VideoAnnotation) and i in a.frames
         ]
+
+        if len(annotations) < 1:
```

Comment:
Are we sure this isn't going to break anything for people using darwin-py? This is a breaking change in behaviour for anyone calling this function themselves, right?

Reply:
This is a breaking change indeed, this will only generate annotations for frames that has them. Removing this we need to change the logic in other places to (perhaps local_dataset) to only load and parse the annotation files that has annotations in them. 

In practice, I think that this is only used in our own training infrastructure, and only for videos. But, happy to discuss other ways to address this instead.